pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -346,6 +347,139 @@ However, as blocks are currently executed sequentially, the
+without occupying any more resources than what was occupied in a first place.
+
+If `stop` policy is set, all running blocks will be stopped and all pending ones
+will be canceled as soon as possible when one block fails.
+
+The `fail_fast` property can have two sub properties, `stop` and `cancel`.
+
+At least one of them is required and if both are stated, `stop` will be
+evaluated first.
+
+Both `stop` and `cancel` properties have only one required property, and that is
+`when` property which holds condition written in
+[Conditions DSL][conditions-reference].
+
+If this condition is fulfilled for given pipeline execution, appropriate
+`fail_fast` policy will be set for blocks of that pipeline.
+
+### An example for setting fail_fast stop policy
+
+In the following example blocks A and B will start immediately and block C will
+wait for block B to be finished.
+
+After 10 seconds when Block A fails, if pipeline was initiated from non-master
+branch, fail_fast `stop` policy will be applied and Block B will be stopped and
+Block C will be canceled.
+
+``` yaml
+version: v1.0
+name: Setting fail_fast stop policy
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+fail_fast:
+  stop:
+    when: "branch != 'master'"
+
+blocks:
+  - name: Block A
+    dependencies: []
+    task:
+      jobs:
+      - name: Job A
+        commands:
+          - sleep 10
+          - failing command
+      - name: Block B
+        dependencies: []
+        task:
+          jobs:
+          - name: Job B
+            commands:
+              - sleep 60
+      - name: Block C
+        dependencies: [Block B]
+        task:
+          jobs:
+          - name: Job C
+            commands:
+              - sleep 60
+```
+
+### An example for setting fail_fast cancel policy
+
+In the following example blocks A and B will start immediately and block C will
+wait for block B to be finished.
+
+After 10 seconds when Block A fails, if pipeline was initiated from non-master
+branch, fail_fast `cancel` policy will be applied.
+
+When this policy is applied, Block C will be canceled since it is not executing
+at that moment, but Block B will be allowed to finish its execution.
+
+``` yaml
+version: v1.0
+name: Setting fail_fast cancel policy
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+fail_fast:
+  cancel:
+    when: "branch != 'master'"
+
+blocks:
+  - name: Block A
+    dependencies: []
+    task:
+      jobs:
+      - name: Job A
+        commands:
+          - sleep 10
+          - failing command
+      - name: Block B
+        dependencies: []
+        task:
+          jobs:
+          - name: Job B
+            commands:
+              - sleep 60
+      - name: Block C
+        dependencies: [Block B]
+        task:
+          jobs:
+          - name: Job C
+            commands:
+              - sleep 60
+```
+
+More examples of conditions for frequent use cases can be found
+[here][when-repo-fail_fast-exemples].
```

Comment:
Please no. :) Users need to get all information they need within the docs website.

Reply:
I will remove this then. 

Those examples are really useful though, and they could eventually be included in some kind of guide or examples collection page in docs, and we can then add link here.

Reply:
Can you include them as part of this iteration?